examples/tutorials/manage-ntp.markdown

```diff
@@ -0,0 +1,741 @@
+
+##### create
+
+```cf3
+      create                => "true",
+```
+
+Valid values for this attribute are `true` or `false` to instruct the agent whether or not to create the file. In other words, the file must exist. If it does not exist, it will be created.
+
+##### perms
+
+```cf3
+      perms                 => mog( "644", "root", "root" ),
+```
+
+This attribute sets the permissions and ownership of the file. `mog()` is function in the CFEngine standard library that sets the `mode`, `owner`, and `group` of the file. In this example, the permissions for the NTP configuration file is set to `644` with owner and group both assigned to `root`.
+
+##### handle
+
+```cf3
+      handle                => "ntp_files_conf",
+```
+
+This is a descriptive handle that will uniquely identify the promise within this policy. The [policy style guide][Policy Style Guide#promise handles] recommends a naming scheme for the handles e.g. `bundle_name_promise_type_class_restriction_promiser`.
+
+##### classes
+
+```cf3
+      classes               => results( "bundle", "ntp_config" );
+```
+The classes attribute here indicates that every time this promise is triggered, whether the file is created, or the content of the file is updated, the class `restart_ntpd` will be set. Recall that this class was also set in the processes promise when the agent determines that the NTP process is not running. Setting this class activates the commands promise which will proactively issue a command to restart the service.
+
+##### template_method
+
+```cf3
+      template_method       => "inline_mustache",
+```
+
+CFEngine supports multiple templating engines, the `template_method` attribute specifies how the promised file content will be resolved. The value `inline_mustache` indicates that we will use the mustache templating engine and specify the template in-line, instead of in an external file.
+
+##### edit_template_string
+
+```cf3
+      edit_template_string  => "$(config_template_string)",
+```
+
+The `edit_template_string` attribute is set to `$(config_template_string)` which holds the mustache template used to render the file content.
+
+##### template_data
+
+```cf3
+      template_data         => mergedata( '{ "driftfile": "$(driftfile)", "servers": servers}' ),
+```
+
+`template_data` is assigned a data container that is in this case constructed by `mergedata()` so that only the necessary data is provided to the template. If `template_data` is not explicitly provided, CFEngine uses `datastate()` by default. It is considered best practice to provide explicit data as this makes it easier to delegate responsibility of the template and that data to different entities where neither are required to know anything about CFEngine itself and it's /much/ more efficient to send the templating engine only the data the template actually uses.
+
+The result of `mergedata()` in the example is equivalent to this json:
```

Comment:
I think it's worth an explanation that `servers` is a bare value and as such it is replaced by an expansion of the `servers` slist.

Reply:
ACK.